### DISCUSSION TITLE
Is it possible to make a port to android

### DISCUSSION THREAD

**Discussion by pleomati** [evilsocket/opensnitch#910] [Development]
great tool for linux,but is it possible to make a port to android which is also linux.Thank you

**Comment by gustavo-iniguez-goya** [comment]
Hi @pleomati ,

Yes, it'd be possible. In fact I tried to port it longlonglooong ago (+-2020). The problem is that we need to compile the `libnetfilter_queue` for android, but I failed to do it at the time.

But you have some similar apps like: [Tracker Control](https://github.com/OxfordHCC/tracker-control-android), [NetGuard](https://github.com/M66B/NetGuard) and [Rethink-app](https://github.com/celzero/rethink-app)

By the way, a student wrote a thesis about this: `DiscoWall: Design And Implementation Of A Firewall For Android Phones`
DiscoWall worked almost *exactly* as opensnitch (but written in C).

The user removed the github user account, so the repository is no longer available. The thesis PDF seems also to be broken: https://disco.cs.uni-kl.de/index.php/teaching/theses/completed-theses

(it's available here: https://silo.tips/download/discowall-design-and-implementation-of-a-firewall-for-android-phones-brian-tewan)

**Reply by pizzadude** [reply]
Also https://github.com/ukanth/afwall (requires rooted device)

**Reply by elovin** [reply]
@pizzadude 
I would prefer a background service which can not be configured on the phone itself e.g. for parental control. 

So running one opensnitch node on the phone as an android service on boot (rooted e.g. lineageos 20 / android 13) and managing it through the GUI on a linux PC on the same local network would be nice (This works with e.g. dnscrypt-proxy but that can not be used to filter ips).
